### PR TITLE
Added documentation for creating submodules.

### PR DESCRIPTION
**Pull request by chriscantu**
Here is the approach I have been using to create submodules for common UI sections like Navbars and alerts.  I am not sure if this is the 'correct' way but thought I would get your feedback.  

If you are cool with this approach, I thought it should be documented.  

Let me know your thoughts.

### CODE REVIEW

File: Readme.md [hunk 1]
```diff
@@ -102,6 +102,30 @@ signInPage.signIn('test user', 'testpassword'); // will navigate to sign in page
...
+var homePage = Page.create({
+    .... // homepage specific components
+});
+
+homePage.navbar = Page.create(navbar);
+module.exports = homePage;
+
+//In Test
+it('logout user', function () {
+    homePage.navbar.logoutLink.click();
```
Comment (Droogans) [inline Readme.md:125]: I like this approach. I think this looks nice, especially for cases where you might have more than one path through a page's features:

``` javascript
it('should create an admin user', function () {
    userPage.create.admin('withPassword');
});

it('should remove a user', function () {
    userPage.remove.user('withUsername');
});
```

Where your directory might look like this:

```
└── test
    └── pages
        ├── user
        │   ├── create.js
        │   └── remove.js
        └── user.js
```

Where the `user.js` file is simply an aggregate of the contents of the `user` directory.

**Comment by chriscantu** [comment]
@stuplum any thoughts on this approach for sub modules?

**Comment by stuplum** [comment]
Modules actually have there own class, although it is more of a factory method. 

homePage.js

``` javascript
var navModule = require('../modules/navModule');

module.exports = BasePage.create({
    url: { value: '/home' },
    navigation: { get: function() { return navModule(this.findElement(this.by.css('.nav'))); } }
});
```

navModule.js

``` javascript
var NavigationModule = Module.extend({
    links:  { get: function() { return this.findElements(this.by.css('a') } },
    logoutLink:  { get: function() { return this.findElements(this.by.css('.logout') } }
});

module.exports = function(context) {
    return new NavigationModule(context);
};
```

So this would alow you to access the logout link using homePage.navigation.logoutLink.click() for example. The selectors for the elements within an element are confined to the outer selector of the module, so you dont have to write the complete selectors for each module item.

eg.

module selector = .nav
link with in module = .logoutLink

instead of:

link within module = .nav .logoutLink

Modules aren't currently documented as they have a fairly large flaw at the moment, which is fundamental to them being useful. They cannot currently be used for selecting multiple modules within a page. If there is only one module within a page ie. a global navigation they are fine.

**Comment by chriscantu** [comment]
@stuplum The way I documented works with multiple sub modules.  If you don't mind this approach for the time being, then you can use the documentation.

Otherwise, we can just close the pull-request.

**Comment by stuplum** [comment]
I'm closing this for the time being, I would like to make the module class work properly

**Pull request closed without merging**